core/vibes/soul/sections/cart/shipping-form/index.tsx

```diff
@@ -0,0 +1,418 @@
+    },
+    onValidate({ formData }) {
+      return parseWithZod(formData, { schema: shippingActionFormDataSchema });
+    },
+    onSubmit(event, { formData }) {
+      event.preventDefault();
+
+      startTransition(() => {
+        formAction(formData);
+        setShowForms(false);
+      });
+    },
+  });
+
+  const formattedAddress = useMemo(() => {
+    const country = Array.isArray(countries)
+      ? countries.find((c) => c.value === state.address?.country)?.label
+      : state.address?.country;
+
+    const city = state.address?.city;
+
+    const statesArray = Array.isArray(states)
+      ? states.find((s) => {
+          if (s.country === state.address?.country) {
+            return true;
+          }
+
+          return false;
+        })
+      : undefined;
+
+    const stateOrProvince = Array.isArray(statesArray?.states)
+      ? statesArray.states.find((s) => s.value === state.address?.state)?.label
+      : state.address?.state;
+
+    const cityAndState = [city, stateOrProvince].filter((item) => item !== undefined).join(', ');
+    const postalCode = state.address?.postalCode ?? '';
+
+    const firstLine = `${cityAndState} ${postalCode}`.trim();
+
+    return (
+      <div>
+        {Boolean(firstLine) && (
+          <>
+            {firstLine} <br />
+          </>
+        )}
+        {country}
+      </div>
+    );
+  }, [
+    countries,
+    state.address?.country,
+    state.address?.city,
+    state.address?.state,
+    state.address?.postalCode,
+    states,
+  ]);
+
+  useEffect(() => {
+    if (addressForm.errors && addressForm.errors.length > 0) {
+      if (state.shippingOptions) {
+        setShowForms(true);
+        setShowAddressForm(true);
+      }
+    }
+  }, [setShowForms, setShowAddressForm, addressForm.errors, state.shippingOptions]);
+
+  useEffect(() => {
+    if (shippingOptionsForm.errors && shippingOptionsForm.errors.length > 0) {
+      setShowForms(true);
+    }
+  }, [setShowForms, shippingOptionsForm.errors]);
+
+  const shippingOptionsControl = useInputControl(shippingOptionsFields.shippingOption);
+  const countryControl = useInputControl(addressFields.country);
+  const stateControl = useInputControl(addressFields.state);
+
+  return (
+    <div className="space-y-4 border-t border-contrast-100 pb-5 pt-4">
+      <div>
+        <div className="flex justify-between">
+          <span>{shippingLabel}</span>
+          {state.shippingOption ? (
+            <span>{state.shippingOption.price}</span>
+          ) : (
+            !showForms && (
+              <button
+                className="font-semibold focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[var(--button-focus,hsl(var(--primary)))] focus-visible:ring-offset-2"
+                onClick={() => setShowForms(!showForms)}
+              >
+                {addLabel}
+              </button>
+            )
+          )}
+        </div>
+
+        {state.shippingOption && (
```

Comment:
@jorgemoya 

We should handle the case where no shipping options are returned:
![image](https://github.com/user-attachments/assets/af2cc475-afe5-4c5a-83bb-2f6b782ae309)

We should display a message stating that `No shipping options are available for the selected region`, or something along those lines.

Reply:
Good call!

Reply:
Implemented as this:

![Screenshot 2025-03-13 at 3 28 18 PM](https://github.com/user-attachments/assets/fea923d1-0909-4d44-bd51-28c04ed570d8)